react/src/lib/components/WebvizDialog/WebvizDialog.tsx

```diff
@@ -0,0 +1,678 @@
+    /**
+     * The name of the action that was called last.
+     */
+    last_action_called?: string;
+};
+
+export type WebvizDialogProps = {
+    /**
+     * The ID used to identify this component in Dash callbacks.
+     */
+    id: string;
+    /**
+     * States if the dialog is open or not.
+     */
+    open?: boolean;
+    /**
+     * Set to true to show dialog as modal
+     */
+    modal?: boolean;
+    /**
+     * The title of the dialog.
+     */
+    title: string;
+    /**
+     * Set the minimum width in px
+     */
+    minWidth?: number;
+    /**
+     * Set the minimum height in px
+     */
+    minHeight?: number;
+    /**
+     * If true, hitting escape will not fire the close callback
+     */
+    disableEscapeKeyDown?: boolean;
+    /**
+     * A list of actions to be displayed as buttons in the lower right corner of the dialog.
+     */
+    actions?: string[];
+    /**
+     * The child elements showed in the dialog
+     */
+    children?: React.ReactNode;
+    /**
+     * Dash-assigned callback that should be called whenever any of the
+     * properties change.
+     */
+    setProps: (parentProps: WebvizDialogParentProps) => void;
+};
+
+export const WebvizDialog: React.FC<WebvizDialogProps> = (props) => {
+    const marginLeft = 16;
+    const marginRight = 16;
+    const marginTop = 16;
+    const marginBottom = 16;
+    const minScrollAreaHeight = 100;
+
+    const [open, setOpen] = React.useState<boolean>(props.open || false);
+    const [actionsCalled, setActionsCalled] = React.useState<number>(0);
+
+    const [dialogPosition, setDialogPosition] = React.useState<Point>(ORIGIN);
+    const [minDialogWidth, setMinDialogWidth] = React.useState<number>(
+        props.minWidth || 0
+    );
+
+    const [dialogSize, setDialogSize] = React.useState<Size | null>(null);
+    const [initialDialogSize, setInitialDialogSize] =
+        React.useState<Size | null>(null);
+    const [initialDialogContentHeight, setInitialDialogContentHeight] =
+        React.useState<number | null>(null);
+
+    const [isMovedOutsideWindow, setIsMovedOutsideWindow] =
+        React.useState<boolean>(false);
+    const [useScrollArea, setUseScrollArea] =
+        React.useState<boolean | undefined>(undefined);
+    const [scrollAreaHeight, setScrollAreaHeight] = React.useState<number>(0);
+
+    const [placeholderDiv, setPlaceholderDiv] =
+        React.useState<HTMLDivElement | null>(null);
+    const wrapperRef = React.useRef<HTMLDivElement>(null);
+    const dialogRef = React.useRef<HTMLDivElement>(null);
+    const dialogTitleRef = React.useRef<HTMLDivElement>(null);
+    const dialogContentRef = React.useRef<HTMLDivElement>(null);
+    const dialogActionsRef = React.useRef<HTMLDivElement>(null);
+
+    const handleSetActive = React.useCallback(() => {
+        const activeDialogs = Array.from(
+            document.getElementsByClassName("WebvizDialog--active")
+        );
+        for (const elm of activeDialogs) {
```

Comment:
`document.getElementsByClassName("WebvizDialog--active")` returns an `HTMLCollection`, which does not contain the `forEach`-method. To use a `forEach` one has to convert to `array` as far as I can see?

Reply:
Yes, that is true but not what I meant. When using `for ... of`, as you do, you do not need the conversion to `Array`. But since we want to use ECMAScript 2016, we want to use `forEach` and do hence need the conversion first. You implemented the conversion to Array which is required for `forEach` but then used `for ... of`.

Reply:
Yes, that's true. Replaced `for ... of` with `forEach`.